### DISCUSSION TITLE
compile

### DISCUSSION THREAD

**Discussion by Marcus89uk** [jagrosh/MusicBot#1075] [Questions] [answered]
how to compile this?

**Comment by jagrosh** [comment, marked as answer]
Compiling JMusicBot from source is not supported.

**Reply by Marcus89uk** [reply]
What do you mean? why have a source if you cannot compile it.

**Reply by jagrosh** [reply]
It is not supported, as in no support or help will be provided for trying to build this yourself.

**Reply by MichailiK** [reply]
Open source does not always mean "includes free support." Most project maintainers assist and guide you in compiling, but some do not. In our case, we have decided against supporting compilation for decreased burden. Experienced Java developers, who can use the source code, are welcome to view or tinker with it.
